src/Aiming.cpp

```diff
@@ -1,18 +1,237 @@
-/*
- * Aiming.cpp
- *
- *  Created on: Jan 22, 2016
- *      Author: Brin Harper
- */
-
-#include <Aiming.h>
-
-Aiming::Aiming() {
-   // TODO Auto-generated constructor stub
-
-}
-
-Aiming::~Aiming() {
-   // TODO Auto-generated destructor stub
-}
-
+/*
+ * Aiming.cpp
+ *
+ *  Created on: Jan 22, 2016
+ *      Author: Brin Harper
+ *
+ * The Aiming class takes in the vision data received from the Raspberry Pi, by way
+ * of the UDP client.  This data is passed to the Aiming class in the form
+ * of an integer array, which represents the coordinates of the target's four corners.
+ * Aiming implements the received vision data in order to align and rotate the robot,
+ * until the the shooter is positioned correctly to shoot at the desired high target.
+ *
+ */
+
+#include <Aiming.h>
+
+Aiming::Aiming(Client* client, DriveTrainController* driveTrainController, DriveStation* driveStation) :
+   m_client(client),
+   m_driveTrainController(driveTrainController),
+   m_driveStation(driveStation)
+{
+   setCurrentState(IDLE);
+   m_currentTargetCoordinates[0] = AimingConstants::targetFlag;
+   for (int i = 1; i <= AimingConstants::numTargetVals; i++) {
+      m_currentTargetCoordinates[i] = 0;
+   }
+
+   lastArrayWasNull = true;
+
+   //Keeps track of the number of blank data sets sent over to Client
+   //This represents the number of action cycles for which the target has not been seen
+   nullArraysInARow = 0;
+}
+
+// IMPORTANT: Call this method to begin aiming process - same as manually setting state to first
+// phase of aiming process
+void Aiming::beginAiming() {
+   setCurrentState(FINDING_TARGET);
+}
+
+// Gives Aiming class access to image data sent over to client from Raspberry Pi
+void Aiming::getNewImageData() {
+
+   //Check for fresh data
+   if(m_client->m_unreadTargetData) {
+
+         // Updates array of current coordinates with data received by client
+         for(int i = 1; i <= AimingConstants::numTargetVals; i++) {
+            m_currentTargetCoordinates[i - 1] = m_client->getTargetData(i);
+         }
+
+         if(m_currentTargetCoordinates[AimingConstants::yUL] == 0) {
+            lastArrayWasNull = true;
+         } else {
+            lastArrayWasNull = false;
+         }
+
+      }
+
+}
+
+
+void Aiming::findTarget() {
+
+   // Rotate while the first coordinate hasn't been found
+   if (lastArrayWasNull) {
+      m_driveTrainController->aimRobotClockwise(5, 0.5);
+
+      std::ostringstream aimingStatus;
+      aimingStatus << "Looking for target";
+      SmartDashboard::PutString("DB/String 8", aimingStatus.str());

```

Comment:
you do not need the whole `std::ostringstream` if you are just printing out a string
`SmartDashboard::PutString("DB/String 8", "Looking for target");` would work